yaml/src/Data/Yaml.hs

```diff
@@ -296,5 +297,5 @@ decodeFileThrow f = liftIO $ decodeFileEither f >>= either throwIO return
 array :: [Value] -> Value
 array = Array . V.fromList
 
-parseMonad :: Monad m => (a -> Parser b) -> a -> m b
+parseMonad :: MonadFail m => (a -> Parser b) -> a -> m b
```

Comment:
In order to avoid breaking backwards compat, I think this change should be conditional on using the new version of base.

Reply:
I disagree with that; conditional APIs are not a good idea. A breaking change is the lesser evil.